packages/legend-application-studio/src/stores/editor-state/element-editor-state/service/ServiceRegistrationState.ts

```diff
@@ -314,5 +318,27 @@ export class ServiceRegistrationState {
         'Service version can not be empty in Semi-interactive and Prod service type',
       );
     }
+
+    // validate service parameter multiplicity. Currently [1..*] is not allowed for service registration.
+    const queryLambda = (this.service.execution as PureExecution).func;
+    const parameters = (queryLambda.parameters ?? []) as object[];
+    if (parameters.length) {
+      const params = buildLambdaVariableExpressions(
+        queryLambda,
+        this.editorStore.graphManagerState,
+      )
+        .filter(filterByType(VariableExpression))
+        .filter(
+          (p) =>
+            p.multiplicity.lowerBound === 1 &&
+            p.multiplicity.upperBound === undefined,
+        );
+      assertTrue(
+        params.length <= 0,
+        `Multiplicy of parameter(s) ${params.map(
+          (p) => p.name,
+        )} is/are [1..*], which is not in the accepted multiplicy list. Please use [0..*], [*..*], [1..1], or [0..1] instead.`,
```

Comment:
Dont think this message is correct.

Reply:
i believe we have a print multplicity function. We could call that on the list we make of TYPICAL_MULTIPLICITY_TYPE